pcdscalc/be_lens_calcs.py

```diff
@@ -925,15 +941,422 @@ def find_z_pos(energy, lens_set, spot_size_fwhm, material=None,
     waist = lam / np.pi * focal_length / w_unfocused
     rayleigh_range = np.pi * waist ** 2 / lam
 
-    logger.info("waist          : %.3e" % waist)
-    logger.info("waist FWHM     : %.3e" % (waist
-                                           * FWHM_SIGMA_CONVERSION / 2.0))
-    logger.info("rayleigh_range : %.3e" % rayleigh_range)
-    logger.info("focal length   : %.3e" % focal_length)
+    logger.info("waist          : %.3e", waist)
+    logger.info("waist FWHM     : %.3e", waist * FWHM_SIGMA_CONVERSION / 2.0)
+    logger.info("rayleigh_range : %.3e", rayleigh_range)
+    logger.info("focal length   : %.3e", focal_length)
 
     w = gaussian_fwhm_to_sigma(spot_size_fwhm) * 2
     delta_z = rayleigh_range * np.sqrt((w / waist) ** 2 - 1)
     z1 = focal_length - delta_z
     z2 = focal_length + delta_z
     z_position = (z1, z2)
     return z_position
+
+
+def plan_set(energy, z_offset, z_range, beam_size_unfocused, size_horizontal,
+             size_vertical=None, exclude=[], max_tot_number_of_lenses=25,
+             max_each=5, focus_before_sample=False):
+    """
+    Macro to help plan for what lens set to use.
+
+    Parameters
+    ----------
+    energy : number
+        Beam energy in KeV
+    z_offset : float
+        Distance from sample to lens_z=0 in meters
+    z_range : list
+        array or tuple of 2 values: minimum and maximum z pos in meters
+    beam_size_unfocused : float
+        Radial size of x-ray beam before focusing in meters
+    size_horiz
+    size_vert
+    excluede : list
+    max_tot_number_of_lenses : int
+    max_each : int
+    focus_before_sample : bool
+
+    Examples
+    --------
+    >>> plan_set(energy=1, z_offset=-10, z_range=[1, 40],
+                           beam_size_unfocused=3, size_horizontal=9,
+                           size_vertical=None, exclude=[],
+                           max_tot_number_of_lenses=1,
+                           max_each=5, focus_before_sample=False)
+
+    N   foc_len/m   Min/um   Max/um   Trans/%  Set
+    0  0.07 466994229.0 2112064677.4   0.0  1 x 50um
+    1  0.14 234997114.5 1057532338.7   0.0  1 x 100um
+    2  0.28 118998557.3 530266169.4   0.0  1 x 200um
+    3  0.43 80332371.5 354510779.6   0.0  1 x 300um
+    4  0.71 49399422.9 213906467.7   0.0  1 x 500um
+    5  1.42 26199711.5 108453233.9   0.0  1 x 1000um
+    6  2.13 18466474.3 73302155.9   0.0  1 x 1500um
+    7  2.84 14599855.7 55726616.9   0.0  1 x 2000um
+    8  4.27 10733237.2 38151078.0   0.0  1 x 3000um
+
+    Where:
+    `foc_len` - is the focal length in um
+    `Min` - Beam full width at half maximum for minimum z position
+    `Max` - Beam full width at half maximum for maximum z position
+    `Trans` - is the lens transmission in %.
+
+   Notes
```

Comment:
There are 3 spaces here instead of 4
```suggestion
    Notes
```

Reply:
Thank you, fixed it in 1643a5d